[DirectX][NFC] Change all DXIL TableGen tokens to CamelCase

bharadwajy opened the pull request:
These changes are in preparation for potential improvement of DXIL operation description and addition of more DXIL operations to `DXIL.td`.

Commit:
[DirectX][NFC] Change all DXIL TableGen tokens to CamelCase

bharadwajy commented:
@llvm-beanz @bogner @python3kgae

dmpots commented on llvm/lib/Target/DirectX/DXIL.td:
```diff
@@ -7,138 +7,139 @@
-  let oload_types = _oload_types;
-  let fn_attr = _fn_attr;
-  let stats_group = _stats_group;
+class DXILOperation<string name, int opCode, DXILOpClass opClass, DXILOpCategory opCategory, string doc,
```
It might be too late given uses in other places, but I think using CamlelCase for `Dxil` instead of all caps `DXIL` (e.g. `DxilOperation<...>`) would be nicer to read.

bharadwajy replied:
> It might be too late given uses in other places, but I think using CamlelCase for `Dxil` instead of all caps `DXIL` (e.g. `DxilOperation<...>`) would be nicer to read.

I certainly considered that - but argued against myself to stick with what did appear to be the usage in other places, as you point out. Additionally, I convinced myself that DXIL is (almost) an acronym for DirectX Intermediate Language (correct?) and hence, CamelCase may not be apt. That said, I remain open to other perspectives on this.

tex3d replied:
Elsewhere (in DXC), we do have an established pattern of using `Dxil` when in CamelCase with other words, and `DXIL` when used by itself.  Perhaps we can establish the preferred pattern here before we have too much code depending on one pattern or the other.

bharadwajy replied:
> Elsewhere (in DXC), we do have an established pattern of using `Dxil` when in CamelCase with other words, and `DXIL` when used by itself. Perhaps we can establish the preferred pattern here before we have too much code depending on one pattern or the other.

It is good to know of the prevailing established pattern. Thanks!

I'll update the PR to follow the same convention in `DXIL.td`.

bogner reviewed (approved):
LGTM

Commit:
Replace DXIL* pattern in CamelCase tokens with Dxil* pattern.
Usage of the acronym DXIL is employed in a freestanding context.

This convention is expected to be followed in DirectX source files
for consistency and readability.

llvm-beanz commented:
> I do not have commit access to `llvm-project` repo. May I request one of the approvers / reviewers to help by pushing the changes? Thanks!

Let's let the buildkite build finish just for posterity, then I'm happy to merge this.

llvm-beanz commented:
The Kite build completed the non-Windows builds and the Windows builders are having trouble keeping up so I'm comfortable merging this.
